lib/esp_sdk/configure.rb

```diff
@@ -1,32 +1,36 @@
 module EspSdk
   class Configure
-    attr_accessor :token, :email, :version, :token_expires_at
+    attr_accessor :token, :email, :version, :token_expires_at, :end_point, :domain
 
     def initialize(options)
-      @email   = options[:email]
-      @version = options[:version] || 'v1'
-      token_setup(options)
+      self.email = options[:email]
+      self.version = options[:version] || 'v1'
+      self.domain = options[:domain]
+      self.token = options[:password] || options[:token]
+      self.end_point = options[:password].present? ? 'new' : 'valid'
+      token_setup
     end
 
-    def uri
-      return @uri if @uri.present?
+    def url(endpoint)
+      "#{domain}/api/#{version}/#{endpoint}"
+    end
 
+    def domain
+      return @domain if @domain.present?
       if EspSdk.production?
-        @uri = 'https://api.evident.io/api'
+        self.domain = 'https://api.evident.io'
       elsif EspSdk.release?
-        @uri = 'https://api-rel.evident.io/api'
+        self.domain = 'https://api-rel.evident.io'
       else
-        @uri = 'http://0.0.0.0:3000/api'
+        self.domain = 'http://0.0.0.0:3000'
       end
     end
 
     private
 
-    def token_setup(options)
-      self.token = options[:password] || options[:token]
-      end_point  = options[:password].present? ? 'new' : 'valid'
+    def token_setup
       client     = Client.new(self)
-      response   = client.connect("#{uri}/#{version}/token/#{end_point}")
+      response   = client.connect(url("token/#{end_point}"))
```

Comment:
That makes this end_point kind of confusing.  I might consider renaming the attr_accessor.